firmware/application/apps/ui_recon.cpp

```diff
@@ -1310,18 +1378,64 @@ size_t ReconView::change_mode(freqman_index_t new_mod) {
+                switch (sampling_rate) {                                 // we use the var fs (sampling_rate) , to set up BPF aprox < fs_max/2 by Nyquist theorem.
+                    case 0 ... 2000000:                                  // BW Captured range  (0 <= 250kHz max )  fs = 8 x 250 kHz
+                        anti_alias_baseband_bandwidth_filter = 1750000;  // Minimum BPF MAX2837 for all those lower BW options.
+                        break;
+                    case 4000000 ... 6000000:                            // BW capture  range (500k ... 750kHz max )  fs_max = 8 x 750kHz = 6Mhz
+                                                                         // BW 500k ... 750kHz   ,  ex. 500kHz   (fs = 8*BW =  4Mhz) , BW 600kHz (fs = 4,8Mhz) , BW  750 kHz (fs = 6Mhz)
+                        anti_alias_baseband_bandwidth_filter = 2500000;  // in some IC MAX2837 appear 2250000 , but both works similar.
+                        break;
+                    case 8800000:  // BW capture 1,1Mhz  fs = 8 x 1,1Mhz = 8,8Mhz . (1Mhz showed slightly higher noise background).
+                        anti_alias_baseband_bandwidth_filter = 3500000;
+                        break;
+                    case 14000000:  // BW capture 1,75Mhz  , fs = 8 x 1,75Mhz = 14Mhz
+                                    // good BPF, good matching, but LCD making flicker , refresh rate should be < 20 Hz , but reasonable picture
+                        anti_alias_baseband_bandwidth_filter = 5000000;
+                        break;
+                    case 16000000:  // BW capture 2Mhz  , fs = 8 x 2Mhz = 16Mhz
+                                    // good BPF, good matching, but LCD making flicker , refresh rate should be < 20 Hz , but reasonable picture
+                        anti_alias_baseband_bandwidth_filter = 6000000;
+                        break;
+                    case 20000000:  // BW capture 2,5Mhz  , fs= 8 x 2,5 Mhz = 20Mhz
+                                    // good BPF, good matching, but LCD making flicker , refresh rate should be < 20 Hz , but reasonable picture
+                        anti_alias_baseband_bandwidth_filter = 7000000;
+                        break;
+                    default:  // BW capture 2,75Mhz, fs = 8 x 2,75Mhz= 22Mhz max ADC sampling) and others.
+                              //  We tested also 9Mhz FPB stightly too much noise floor, better 8Mhz
+                        anti_alias_baseband_bandwidth_filter = 8000000;
+                }
+                record_view->set_sampling_rate(sampling_rate);
+                receiver_model.set_sampling_rate(sampling_rate);
+                receiver_model.set_baseband_bandwidth(anti_alias_baseband_bandwidth_filter);
+            };
+            text_ctcss.set("        ");
             break;
         default:
             break;
     }
+    if (new_mod != SPEC_MODULATION)
+        record_view->set_sampling_rate(recording_sampling_rate);
```

Comment:
Or maybe record_view is already setting the baseband sample rate?

Reply:
Spec is setting it from inside a on_change button, and recording_sampling_rate in that case is local to the on_change button.
Else it should use the recording_sampling_rate set in the other modes, so here we are with a pesky condition.
I'll comment/change one of them to another name